dnsrecon.py

```diff
@@ -1271,12 +1271,23 @@ def ds_zone_walk(res, domain):
             records.extend(lookup_next(hostname, res))
 
             # Arrange the arguments for the transformations
-            fields = re.search("(^[^.]*).(\S*)", hostname)
-            params = [hostname, fields.group(1), fields.group(2)]
+            fields = re.search("^(^[^.]*)\.(\S+\.\S*)$", hostname)
+
+            domain_portion = hostname
+            if fields and fields.group(2):
+                domain_portion = fields.group(2)
+
+            host_portion = ""
+            if fields and fields.group(1):
+                host_portion = fields.group(1)
+
+            params = [hostname, host_portion, domain_portion]
 
             for transformation in transformations:
```

Comment:
Is there value to continuing to apply transformations if the code is already successfully walking the domain?